libs/MeshKernel/src/Mesh2DIntersections.cpp

```diff
@@ -0,0 +1,363 @@
+    double adimensionalEdgeDistance = constants::missing::doubleValue;
+    const auto polyLineSize = static_cast<UInt>(polyLine.size());
+
+    const auto checkPolyLineIndex = [&]
+    {
+        // forward
+        if (direction == 1)
+        {
+            return firstIndex < polyLineSize - 2;
+        }
+        // backward
+        if (direction == -1)
+        {
+            return firstIndex >= 1;
+        }
+        return false;
+    };
+
+    while (!intersectionFound && checkPolyLineIndex() && numSteps < maxSearchSegments)
+    {
+        // forward
+        if (direction == 1)
+        {
+            firstIndex = secondIndex;
+            secondIndex = firstIndex + 1;
+        }
+
+        // backward
+        if (direction == -1)
+        {
+            secondIndex = firstIndex;
+            firstIndex = firstIndex - 1;
+        }
+
+        auto intersection = AreSegmentsCrossing(polyLine[firstIndex],
+                                                polyLine[secondIndex],
+                                                m_mesh.m_nodes[m_mesh.m_edges[edgeIndex].first],
+                                                m_mesh.m_nodes[m_mesh.m_edges[edgeIndex].second],
+                                                false,
+                                                m_mesh.m_projection);
+
+        intersectionFound = std::get<0>(intersection);
+        crossProductValue = std::get<2>(intersection);
+        adimensionalPolylineSegmentDistance = std::get<3>(intersection);
+        adimensionalEdgeDistance = std::get<4>(intersection);
+
+        numSteps++;
+    }
+
+    return {intersectionFound, firstIndex, secondIndex, crossProductValue, adimensionalPolylineSegmentDistance, adimensionalEdgeDistance};
+}
+
+void Mesh2DIntersections::Compute(const std::vector<Point>& polyLine)
+{
+    // 1. Find the intersection of any segment of the polyline with the mesh return if nothing is found
+    std::ranges::fill(m_edgesIntersectionsCache, EdgeMeshPolylineIntersection());
+    std::ranges::fill(m_facesIntersectionsCache, FaceMeshPolylineIntersection());
+
+    const auto polyLineSize = static_cast<UInt>(polyLine.size());
+
+    std::vector<double> cumulativeLength(polyLine.size(), 0.0);
+    for (UInt i = 1; i < polyLineSize; ++i)
+    {
+        cumulativeLength[i] = cumulativeLength[i - 1] + ComputeDistance(polyLine[i], polyLine[i - 1], m_mesh.m_projection);
+    }
+
+    std::queue<std::array<UInt, 3>> crossingEdges;
+    std::vector<bool> vistedEdges(m_mesh.GetNumEdges(), false);
+    std::vector<bool> vistedFace(m_mesh.GetNumEdges(), false);
+
+    // keep traversing the polyline as long crossed edges are found
+    while (true)
+    {
+        // find the seed
+        const auto [isSeedFound, crossedEdgeIndex, crossedSegmentIndex] = GetIntersectionSeed(
+            m_mesh,
+            polyLine,
+            vistedEdges);
+
+        // no valid seed found in the entire mesh, we are done
+        if (!isSeedFound)
```

Comment:
As above, may be able to remove isSeedFound. It is just a proxy for crossedEdgeIndex == nullvalue

Reply:
done, although it exposes the crossedEdgeIndex invalid value